azkaban-common/src/test/java/azkaban/user/XmlUserManagerTest.java

```diff
@@ -150,21 +156,82 @@ public void testAutoReload() throws Exception {
         if (!user8.getUserId().equals("user8")) {
-          System.out
-              .println("Failed to get correct user. Expected user8, got " + user8.getUserId());
           fail("Failed to get correct user. Expected user8, got " + user8.getUserId());
         }
-        System.out.println("Config reloaded successfully.");
+        log.info("Config reloaded successfully.");
       } catch (final UserManagerException e) {
-        System.out.println("Test failed " + e.getMessage());
-        fail("Test failed " + e.getMessage());
+        fail("Test failed " + e.toString());
       }
-    } catch (ConditionTimeoutException te) {
-      System.out.println("The config did not reload in 60 seconds");
-      fail("The config did not reload in 60 seconds");
+    } catch (final ConditionTimeoutException te) {
+      fail("The config did not reload in 30 seconds");
     } finally {
-      // Revert the file back
-      Files.write(Paths.get(path), origLines);
+      // Delete the file
+      Files.delete(filePath);
+    }
+  }
+
+  /**
+   * Test auto reload of user XML
+   */
+  @Test
+  public void testAutoReloadFail() throws Exception {
+    final Props props = new Props(this.baseProps);
+    final URL configURL = Resources.getResource("test-conf/azkaban-users-test1.xml");
+    final String origpath = configURL.getPath();
+    // Copy the file to keep original file unmodified
+    final String path = origpath.replace("test1", "test1_auto_reload_fail");
+    final Path filePath = Paths.get(path);
+    Files.copy(Paths.get(origpath), filePath, StandardCopyOption.REPLACE_EXISTING);
+    // sleep for a second to have different modification time.
+    Thread.sleep(1000L);
+    props.put(XmlUserManager.XML_FILE_PARAM, path);
+
+    final UserManager manager;
+    try {
+      manager = new XmlUserManager(props);
+    } catch (final RuntimeException e) {
+      fail("Should have found the xml file");
+      return;
+    }
+
+    // Get the user8 from existing XML with password == password8
+    final User user8 = manager.getUser("user8", "password8");
+
+    // Modify the password for user8
+    // TODO : djaiswal : Find a better way to modify XML
+    final FileTime origModifiedTime = Files.getLastModifiedTime(filePath);
+    log.info("File modification time = " + origModifiedTime.toString());
+
+    // Make sure the file gets reverted back.
+    try {
+      // Update the file to make it empty.
+      Files.write(filePath, new byte[0], StandardOpenOption.TRUNCATE_EXISTING);
+      final FileTime lastModifiedTime = Files.getLastModifiedTime(filePath);
+      log.info("File modification time after write = " + lastModifiedTime.toString());
+      if (origModifiedTime.equals(lastModifiedTime)) {
+        // File did not update
+        fail("File did not update.");
+      }
+      // Try for 30 seconds polling every 10 seconds if the config is reloaded
+      Awaitility.await().atMost(30L, TimeUnit.SECONDS).
```

Comment:
Whoa, so this is adding a fixed +30s to the test suite runtime? Maybe this should be disabled or then optimize to run without any actual sleeping.

Reply:
Awaitility can check for a condition -- @djaiswal83 , can you add this, so that it will continue as soon as the condition is satisfied.